contrib/postprocessing/lethe_pyvista_tools.py

```diff
@@ -675,27 +833,43 @@ def mixing_index_nearest_neighbors(self, n_neighbors = 15, reference_array = "pa
         # "Characterizing solids mixing in DEM simulations." cell 1 (2007).
 
         # If neighbors is not an attribute of the dataframe
-        if hasattr(self.df[0], "neighbors") == False or len(self.df[0].neighbors[0]) != n_neighbors:
+
+        if self.df_available:
+            df = self.df[0]
+        else:
+            df = self.get_df(0)
+
+        if hasattr(df, "neighbors") == False or len(df['neighbors'][0]) != n_neighbors:
             self.get_nearest_neighbors(n_neighbors = n_neighbors)
 
         # Create empty list to store mixing_index per time-step
         self.mixing_index = []
         self.mixing_index_std = []
 
         # Loop through dataframes and find its mixing index
-        pbar = tqdm(total = len(self.df), desc = "Calculating mixing index")
-        for i in range(len(self.df)):
+        pbar = tqdm(total = len(self.list_vtu), desc = "Calculating mixing index")
+        for i in range(len(self.list_vtu)):
+
+            if self.df_available:
+                df = self.df[i]
+            else:
+                df = self.get_df(i)
```

Comment:
Everytime get_df is called it needs to read the data? Wouldn't it be better to save the current data of the vtu?

Reply:
Yes. It needs to be like this to make sure the right df is called.